web/src/lib/components/pages/record/record.tsx

```diff
@@ -130,7 +132,11 @@ export default class RecordPage extends Component<RecordProps, RecordState> {
 
   private processRecording(info: AudioInfo) {
     let recordings = this.state.recordings;
-    recordings.push(info);
+    if (this.state.isReRecord) {
+      recordings.splice(this.state.reRecordIndex, 0, info);
+    } else {
+      recordings.push(info);
+    }
 
     this.setState({
```

Comment:
in this `setState`, where `isReRecord` is reset to `false`, `reRecordIndex` should also be reset to its default (-1).

other than that, LGTM :)

@Gregoor @mikehenrty please check as well ;)